File: 169-majority-element/169. Majority Element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length,c=0,ans=0;
        for(int i=0; i<n;i++){
            if(c==0){
                c++;
                ans=nums[i];
            }
            else if(nums[i]==ans){
                c++;
            }
            else
                c--;
        }
        return ans;
    }
}

File: 169-majority-element/169. Majority Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        c = 0
        e = 0
        for i in range(n):
            if(c==0):
                c+=1
                e=nums[i]
            elif(nums[i]==e):
                c+=1
            else:
                c-=1
        return e